awareness.js

```diff
@@ -189,9 +237,10 @@ export const removeAwarenessStates = (awareness, clients, origin) => {
 /**
  * @param {Awareness} awareness
  * @param {Array<number>} clients
+ * @param {typeof DefaultAwarenessStateEncoder|typeof BinaryAwarenessStateEncoder} stateEncoder The encoder to use for encoding and decoding each state entry
```

Comment:
I don't use jsdoc types a lot, but my understanding is that `typeof Blah` is the best way to describe a constructor.

Another way to type this would be to instead type the "interface" expected - but that would make changes to the requirements of the encoder a breaking change.